frontend/src/reducers/authReducer.js

```diff
@@ -0,0 +1,59 @@
+import {
+    USER_LOADED,
+    USER_LOADING,
+    AUTH_ERROR,
+    LOGIN_SUCCESS,
+    LOGIN_FAIL,
+    LOGOUT_SUCCESS,
+    REGISTER_SUCCESS,
+    REGISTER_FAIL,
+} from '../actions/types';
+
+const initialState = {
+    token: localStorage.getItem('token'),
+    isAuthenticated: null,
+    isLoading: false,
+    user: null,
+};
+
+
+export default function(state=initialState, action) {
+    switch(action.type) {
+        case USER_LOADING:
+            // Get user from backend
+            return {
+                ...state,
+                isLoading: true
```

Comment:
IS there going to have a loading page? Please explain a bit why we need this prop in state

Reply:
it allows for us to support loading if needed. I added it if we want to have loading spinners and stuff for styling. It is not currently used but is there for extensibility